internal/provision/devices.go

```diff
@@ -32,73 +38,28 @@ import (
 )
 
 func LoadDevices(path string, dic *di.Container) errors.EdgeX {
-	if path == "" {
-		return nil
-	}
-
-	absPath, err := filepath.Abs(path)
-	if err != nil {
-		return errors.NewCommonEdgeX(errors.KindServerError, "failed to create absolute path", err)
-	}
-
-	files, err := os.ReadDir(absPath)
-	if err != nil {
-		return errors.NewCommonEdgeX(errors.KindServerError, "failed to read directory", err)
-	}
+	var addDevicesReq []requests.AddDeviceRequest
 
-	if len(files) == 0 {
+	if path == "" {
 		return nil
 	}
 
 	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
-	lc.Infof("Loading pre-defined devices from %s(%d files found)", absPath, len(files))
-
-	var addDevicesReq []requests.AddDeviceRequest
 	serviceName := container.DeviceServiceFrom(dic.Get).Name
-	for _, file := range files {
-		var devices []dtos.Device
-		fullPath := filepath.Join(absPath, file.Name())
-		if strings.HasSuffix(fullPath, yamlExt) || strings.HasSuffix(fullPath, ymlExt) {
-			content, err := os.ReadFile(fullPath)
-			if err != nil {
-				lc.Errorf("Failed to read %s: %v", fullPath, err)
-				continue
-			}
-			d := struct {
-				DeviceList []dtos.Device `yaml:"deviceList"`
-			}{}
-			err = yaml.Unmarshal(content, &d)
-			if err != nil {
-				lc.Errorf("Failed to YAML decode %s: %v", fullPath, err)
-				continue
-			}
-			devices = d.DeviceList
-		} else if strings.HasSuffix(fullPath, ".json") {
-			content, err := os.ReadFile(fullPath)
-			if err != nil {
-				lc.Errorf("Failed to read %s: %v", fullPath, err)
-				continue
-			}
-			err = json.Unmarshal(content, &devices)
-			if err != nil {
-				lc.Errorf("Failed to JSON decode %s: %v", fullPath, err)
-				continue
-			}
-		} else {
-			continue
+	parsedUrl, err := url.Parse(path)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindServerError, "failed to parse url", err)
```

Comment:
Need more context here.
```suggestion
		return errors.NewCommonEdgeX(errors.KindServerError, "failed to parse devices path as a url", err)
```